### PR TITLE
[master] Adding subscription tests

### PR DESCRIPTION
**Pull request by bzawisto**
## Description
This fixes missing events sent via eth_subscribe - we should use the same type of receipt for both isolated server and full node.

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/libServer/IsolatedServer.cpp [hunk 1]
```diff
@@ -1150,10 +1150,12 @@ void IsolatedServer::PostTxBlock() {
       cacheUpdate.StartEpoch(epoch, blockHash, 0, txnHashes.size());
       Json::Value receipt;
       for (const auto& tx : txnHashes) {
-        if (!ExtractTxnReceipt(tx, receipt)) {
-          LOG_GENERAL(WARNING, "Extract txn receipt failed for " << tx);
-        }
-        cacheUpdate.AddCommittedTransaction(epoch, 0, tx, receipt);
+        TxBodySharedPtr tptr;
```
Comment (valeryz) [inline src/libServer/IsolatedServer.cpp:1153]: Does LookupServer need something similar?

**Review by valeryz — APPROVED** [review]
Also in LookupServer?

File: tests/EvmAcceptanceTests/helper/GeneralHelper.js [hunk 2]
```diff
@@ -18,6 +18,10 @@ var general_helper = {
     return hre.network.config.url;
   },
 
+  getWebsocketUrl: function () {
+    return hre.network.config.websocketUrl;
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/helper/GeneralHelper.js:22]: where did you get it? I mean `websocketUrl` is it a custom field? if so, you should add it to hardhat.config.js as well and skip these tests for non-zilliqa networks

File: tests/EvmAcceptanceTests/test/Subscriptions.js [hunk 3]
```diff
@@ -0,0 +1,758 @@
+const {expect} = require("chai");
+const {ethers} = require("hardhat");
+
+const general_helper = require("../helper/GeneralHelper");
+
+// Ensures all events in current eventloop run are dispatched
+async function waitForEvents(events, timeout = 0) {
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/Subscriptions.js:7]: let's not pollute the global scope. please add this function to `this` in `before` block. then you can use it like `this.waitForEvents` all over the tests

File: tests/EvmAcceptanceTests/test/Subscriptions.js [hunk 4]
```diff
@@ -0,0 +1,758 @@
...
+
+      // Deliberately close subscription
+      secondEventsContract.removeAllListeners();
+
+      const VECTORS = [
+        [senderAddress, "0x05A321d0B9541Ca08d7e32315Ca186cC67A1602c", new ethers.BigNumber.from(100)],
+        [senderAddress, "0xF0Cb24aC66ba7375Bf9B9C4Fa91E208D9EAAbd2e", new ethers.BigNumber.from(200)]
+      ];
+
+      for (let i = 0; i < VECTORS.length; ++i) {
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/Subscriptions.js:748]: `forEach`...

File: tests/EvmAcceptanceTests/test/Subscriptions.js [hunk 5]
```diff
@@ -0,0 +1,758 @@
...
+        const [_SENDER, DEST, AMOUNT] = VECTORS[i];
+        await expect(contract.event3(DEST, AMOUNT)).to.emit(contract, "Event3");
+      }
+      receivedEvents = await waitForEvents(receivedEvents);
+      expect(receivedEvents).to.have.length(1);
+    });
+
+    it("Should receive event with complex scenario - 2 aaa", async function () {
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/Subscriptions.js:618]: change aaa :D

File: tests/EvmAcceptanceTests/test/Subscriptions.js [hunk 6]
```diff
@@ -0,0 +1,758 @@
...
+  describe("When event is triggered with zero indexed parameters", function () {
+    it("Should receive event regardless of provided filters", async function () {
+      let receivedEvents = [];
+      const filter = eventsContract.filters.Event0();
+      eventsContract.on(filter, (from, to, amount, _event) => {
+        receivedEvents.push([from, to, amount]);
+      });
+
+      const VECTORS = [
+        [senderAddress, "0x05A321d0B9541Ca08d7e32315Ca186cC67A1602c", new ethers.BigNumber.from(100)],
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/Subscriptions.js:37]: what are these fixed addresses? Can't you get them from accounts?

**Pull request merged**